test/integration/BUILD

```diff
@@ -46,6 +51,33 @@ envoy_cc_test(
     ],
 )
 
+envoy_cc_test(
+    name = "ads_integration_test",
+    srcs = ["ads_integration_test.cc"],
+    deps = [
+        ":ads_integration_lib",
+        ":http_integration_lib",
+        "//source/common/config:protobuf_link_hacks",
+        "//source/common/config:resources_lib",
+        "//source/common/protobuf:utility_lib",
+        "//source/extensions/transport_sockets/tls:config",
+        "//source/extensions/transport_sockets/tls:context_config_lib",
+        "//source/extensions/transport_sockets/tls:context_lib",
+        "//source/extensions/transport_sockets/tls:ssl_socket_lib",
+        "//test/common/grpc:grpc_client_integration_lib",
+        "//test/mocks/runtime:runtime_mocks",
+        "//test/mocks/server:server_mocks",
+        "//test/test_common:network_utility_lib",
+        "//test/test_common:utility_lib",
+        "@envoy_api//envoy/api/v2:cds_cc",
+        "@envoy_api//envoy/api/v2:discovery_cc",
+        "@envoy_api//envoy/api/v2:eds_cc",
+        "@envoy_api//envoy/api/v2:lds_cc",
+        "@envoy_api//envoy/api/v2:rds_cc",
+        "@envoy_api//envoy/service/discovery/v2:ads_cc",
```

Comment:
Thank you! Removed unnecessary ones, but I think the library does need the API since the return values of build* are envoy::api::v2::Listener/Cluster, right? Technically they're included transitively through the config dump proto though. Not sure if guidelines are generally to explicitly include them.